### DISCUSSION TITLE
How to update source state

### DISCUSSION THREAD

**Discussion by dpetka2001** [twpayne/chezmoi#1778] [Q&A] [answered]
So i would like to ask which is the correct way of making changes. I recently removed a file from a target directory i 
had added to chezmoi (I just deleted the file in the destination directory, aka my home dir) and was wondering if I could update the source state to match my current changes. Or is it better practice to remove the file in the source state and then 
also apply the changes to my home dir?

**Comment by twpayne** [comment, marked as answer]
If you do not remove the file from the source state then chezmoi will re-create in the target directory next time you run `chezmoi apply`.

To remove a file from both the source state and the target directory you can use [`chezmoi remove`](https://github.com/twpayne/chezmoi/blob/master/docs/REFERENCE.md#remove-target).

**Reply by dpetka2001** [reply]
So `target` can be even a single file? I thought that `target` is what i say chezmoi should manage. For example `chezmoi add ~/.config/nvim` then the target would be nvim dir and subdirs (thus i assumed that every file within these dirs could not be used as a target, but apparently i misunderstood?) . I can explicitly use any file as `target`? Btw, regarding my initial question, i got around it with just re-adding the target after deleting the file in destination directory and that also updated the source state in source directory.

**Reply by twpayne** [reply]
Yes, chezmoi manages individual files, i.e. individual files like `~/.bashrc` can be targets.

**Reply by dpetka2001** [reply]
Thank you so much for the clarification :)
